addons/implied-repos/common/src/main/java/org/commonjava/indy/implrepo/change/ImpliedRepositoryDetector.java

```diff
@@ -459,6 +472,43 @@ else if ( !config.isIncludeSnapshotRepos() && !repo.isReleasesEnabled() )
                 else
                 {
                     logger.debug( "Found existing RemoteRepositories: {}", rrs );
+
+                    for ( final RemoteRepository rr : rrs )
+                    {
+                        rr.setMetadata( METADATA_ORIGIN, IMPLIED_REPO_ORIGIN );
+                        try
+                        {
+                            rr.setMetadata( IMPLIED_BY_STORES, mapper.writeValueAsString(
+                                            new ImpliedRepoMetadataManager.ImpliedRemotesWrapper(
+                                                            Collections.singletonList( job.store.getKey() ) ) ) );
+                        }
+                        catch ( JsonProcessingException e )
+                        {
+                            logger.error( "Failed to set {}", IMPLIED_BY_STORES );
+                            continue;
+                        }
+                        rr.setPathMaskPatterns( ref.getPathMaskPatterns() );
```

Comment:
We may need to aggregate pathMaskPatterns, just like the job key, when the repo is pre-existing.

The other thing I'm not sure about is whether `getRemoteRepositoriesByUrl()` will return disabled repositories. If so, we should attempt to re-validate the repo and re-enable it if it works.

Reply:
Aggregating the job key makes sense, but for the pathMaskPatterns, is the merging possible to adding extra restrictions for the targeting group? I guess that's why we need to  check it earlier that only filter the ones with exactly the same pathMaskPatterns.

Reply:
Regarding disabled repositories I disagree. If the pre-existing repo is disabled, it's usually for a reason and re-enabling it automatically can cause more issues.
And regarding pathMaskPatterns - don't merge. please. If there are several instances with different patterns, it's also for a reason. E.g. we have mrrc-ga filtered to only community versions and mrrc-ga-rh filtered to only redhat versions and those are used in different groups. If implied repos addon changed the configuration for either of them just because it is referenced in a pom it would break PNC artifact resolving.

Reply:
As above, i just merged the store keys but not the pathMaskPatterns. that should be fine i guess, right?